krun/cpu.py

```diff
@@ -0,0 +1,117 @@
+# Various CPU related abstrations
+
+import time
+import os
+from krun.detect import detect_platform, PLATFORM_LINUX
+
+# Checking the CPU is running at full-speed
+# -----------------------------------------
+# Typically you need to be root to change the power settings, so we just
+# crash out and let the user sort it out in the case that the CPU is not
+# screaming along at max speed.
+
+LINUX_GOV_FILE = "/sys/devices/system/cpu/cpu0/cpufreq/scaling_governor"
+
+class CPUNotThrottledError(Exception):
+    pass
+
+def check_cpu_throttled():
+    print("Checking CPU is throttled")
+    p = detect_platform()
+    if p == PLATFORM_LINUX:
+        check_cpu_throttled_linux()
+    else:
+        raise NotImplementedError("I don't know how to check your CPU")
+
+def check_cpu_throttled_linux():
+    with open(LINUX_GOV_FILE, "r") as fh:
+        v = fh.read().strip()
+
+    if v != "performance":
+        raise CPUNotThrottledError(("Expected 'performance' got '%s'. "
+                                    "Use cpufreq-set from the "
+                                    "cpufrequtils package") % v)
```

Comment:
Given this error will be seen by end users, wouldn't it be better to do something like
```python
  sys.stderr.write("Expected ... Use cpufreq-set ...")
  sys.exit(1)
```

(where ... is the full message from the exception).